tests/test_plugins.py

```diff
@@ -0,0 +1,140 @@
+
+    def test_track_plugin(self):
+        plugin = add_plugin(
+            placeholder=self.placeholder,
+            plugin_type=AudioTrackPlugin.__name__,
+            language=self.language,
+        )
+        self.assertEqual(plugin.plugin_type, "AudioTrackPlugin")
+
+    def test_plugin_structure(self):
+        audio_file = get_filer_file("test_file.mp3")
+        track_file = get_filer_file("test_track.vtt")
+        request_url = self.page.get_absolute_url(self.language) + "?toolbar_off=true"
+
+        parent = add_plugin(
+            placeholder=self.placeholder,
+            plugin_type=AudioPlayerPlugin.__name__,
+            language=self.language,
+            template="default",
+        )
+        self.page.publish(self.language)
+        self.assertEqual(parent.get_plugin_class_instance().name, "Audio player")
+
+        with self.login_user_context(self.superuser):
+            response = self.client.get(request_url)
+
+        self.assertIn(b"No audio file available.", response.content)
+
+        child = add_plugin(
+            target=parent,
+            placeholder=self.placeholder,
+            plugin_type=AudioFilePlugin.__name__,
+            language=self.language,
+            audio_file=audio_file,
+        )
+        self.page.publish(self.language)
+        self.assertEqual(child.audio_file.label, "test_file.mp3")
+
+        with self.login_user_context(self.superuser):
+            response = self.client.get(request_url)
+
+        self.assertIn(b"<audio controls", response.content)
+        self.assertNotIn(b"No audio file available.", response.content)
+        self.assertContains(response, audio_file.label)
+
+        track = add_plugin(
+            target=child,
+            placeholder=self.placeholder,
+            plugin_type=AudioTrackPlugin.__name__,
+            language=self.language,
+            kind="subtitles",
+            src=track_file,
+            srclang=self.language,
+        )
+        self.page.publish(self.language)
+        self.assertEqual(track.src.label, "test_track.vtt")
+
+        with self.login_user_context(self.superuser):
+            response = self.client.get(request_url)
+
+        self.assertIn(b"<track kind", response.content)
+        self.assertNotIn(b"No audio file available.", response.content)
+        self.assertContains(response, track_file.label)
+
+        folder = add_plugin(
+            target=parent,
+            placeholder=self.placeholder,
+            plugin_type=AudioFolderPlugin.__name__,
+            language=self.language,
+            audio_folder=get_filer_folder(),
+        )
+        self.page.publish(self.language)
+        self.assertEqual(folder.audio_folder.name, "test_folder")
+
+        with self.login_user_context(self.superuser):
+            response = self.client.get(request_url)
+
+        self.assertContains(response, "No matching audio files were found in the specified folder.")
+
+        # cleanup
+        audio_file.delete()
```

Comment:
should happen in the tear down

Reply:
Did that on purpose, don't want to create a second file for each run and then delete it, for double checking I've got the cleanup lines:

https://github.com/divio/djangocms-audio/pull/20/files#diff-327b130c82fa4e83e97f6c1a50a0da1cR32